src/components/Provider.js

```diff
@@ -1,62 +1,35 @@
-import React, { Component } from 'react'
+import React, { useMemo, useEffect } from 'react'
 import PropTypes from 'prop-types'
 import { ReactReduxContext } from './Context'
 import Subscription from '../utils/Subscription'
 
-class Provider extends Component {
-  constructor(props) {
-    super(props)
-
-    const { store } = props
-
-    this.notifySubscribers = this.notifySubscribers.bind(this)
+function Provider({ store, context, children }) {
+  const contextValue = useMemo(() => {
     const subscription = new Subscription(store)
-    subscription.onStateChange = this.notifySubscribers
-
-    this.state = {
+    subscription.onStateChange = subscription.notifyNestedSubs
```

Comment:
This is what I determined `subscription.onStateChange = this.notifySubscribers` to actually boiled down to.  I'm wondering if `subscription` couldn't just handle this internally?

Reply:
We use the same Subscription class for the `connect` components, where this is part of our top-down update mechanic. So, it needs to stay configurable. This is just a quirk of re-use.